packages/ui-components/src/components/relative-time-picker/relative-time-picker.types.ts

```diff
@@ -0,0 +1,102 @@
+import { EventEmitter } from '@stencil/core';
+import { DateInput, ICalendarAdvanceTime } from '../../types';
+
+export interface IRelativeTimePicker {
+	/** (optional) Selected time range key */
+	selectedTimeKey?: string;
+	/** (optional) Selectable relative time options */
+	options?: IRelativeTimePickerOption[][];
+	/** (optional) Currently selected timezone name */
+	selectedTimezone?: string;
+	/** (optional) List of all selectable timezones */
+	timezones?: string[];
+	/** (optional) Defines if the customize interval select option is available */
+	customizeIntervalOptionVisible?: boolean;
+	/** (optional) Defines if the timezone select option is available */
+	timezoneSelectVisible?: boolean;
+}
+
+export interface IRelativeTimePickerEvents {
+	/** Emitted when the selected time key changes */
+	selectedRelativeTimeChange: EventEmitter<ICalendarAdvanceTime>;
+	/** Emitted when customize interval is clicked */
+	customizeIntervalClicked: EventEmitter<string>;
+	/** Emitted when selected timezone changes */
+	timezoneChange: EventEmitter<string>;
+}
+
+export interface IRelativeTimePickerOption {
+	/** (required) Option label ex: Last Week, Last 24h */
+	label: string;
+	/** (required) value of the option selected */
+	value: string;
+	/** (required) Specifies the type of relative time config to be considered for the time calculation */
+	comparisonConfig: ERelativeTimeComparisonConfig;
+	/** (optional) used to format the time range label description ex: 14 Feb until 12:39 */
+	labelRangeFormatter?: IDateTimeRangeFormatter;
+	/** (optional) start date (Date, Time or DateTime can be defined with the dateTimeFormatter) */
+	startDate?: DateTimeCalculation;
+	/** (optional) end date (Date, Time or DateTime can be defined with the dateTimeFormatter) */
+	endDate?: DateTimeCalculation;
+}
+
+export type DateTimeCalculation = {
+	// Date which the calculation should be relative to.
+	// if undefined, now will be used
+	date?: DateInput;
+	// the amount of units to add to the relative time
+	// if this number is negative, the amount will be subtracted
+	amount?: number;
+	// the unit type to add/subtract to the relative time
+	unit?: string;
+};
+
+export interface IDateTimeRangeFormatter {
+	/** Defines how the initial date is formatted on the relative time select description
+	 * 	If empty, now start date will be displayed
+	 */
+	startDateFormatter?: string;
```

Comment:
do we have any specific way to format these strings? or is this not the purpose of this?

Reply:
Yeah, dayJS package provides a method to format a date according with a string. Here we have three formatting strings because in some cases, the start date format is different from the end date format, for example: 13 Feb to 14 Feb 20:30. Since the separator (to, until, ...) can change from option to option just like the start and end date format, I opted by having three different formatters that are concatenated to form the description shown above (ex: 10 Mar until 00:02). 
The formatter accepts determinate date parsing chars like MMM -> Jan, Feb, Mar or MM -> 01, 02, 03. We can have multiple options here.

Reply:
Thanks for the explanation! :)